module,execution_time (sec),success,timestamp,error,error_short
cowidev.vax.batch.argentina,16.18,True,2023-06-11T07:00:22,,
cowidev.vax.batch.australia,,,2023-06-11T07:00:06,,
cowidev.vax.batch.austria_old,,,2023-06-11T07:00:06,,
cowidev.vax.batch.belgium,9.99,True,2023-06-11T07:00:16,,
cowidev.vax.batch.bolivia,4.82,True,2023-06-11T07:00:11,,
cowidev.vax.batch.canada,5.1,False,2023-06-11T07:01:57,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 282, in main
    Canada().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 270, in export
    df = df.pipe(self.pipeline, df_man)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 250, in pipeline
    .pipe(self.pipe_make_monotonic)[
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/canada.py"", line 230, in pipe_make_monotonic
    df = df.pipe(self.make_monotonic, max_removed_rows=self.max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
",1 chunks of lengths 36 would be NaNed for metric people_vaccinated. That is more than maximum allowed (22) by make_monotonic() - check the data. Check dates ['2023-01-26']. Location(s) affected: {'Canada'}
cowidev.vax.batch.chile,1.02,True,2023-06-11T07:00:07,,
cowidev.vax.batch.czechia,25.72,True,2023-06-11T07:00:31,,
cowidev.vax.batch.denmark,,,2023-06-11T07:00:06,,
cowidev.vax.batch.ecdc,88.6,True,2023-06-11T07:01:34,,
cowidev.vax.batch.ecuador,,,2023-06-11T07:00:06,,
cowidev.vax.batch.estonia,0.76,True,2023-06-11T07:00:06,,
cowidev.vax.batch.france,9.03,True,2023-06-11T07:00:15,,
cowidev.vax.batch.germany,0.22,True,2023-06-11T07:00:06,,
cowidev.vax.batch.greece,24.41,True,2023-06-11T07:00:30,,
cowidev.vax.batch.hong_kong,4.6,False,2023-06-11T07:02:01,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/hong_kong.py"", line 224, in main
    HongKong().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/hong_kong.py"", line 208, in export
    df = df_base.pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/hong_kong.py"", line 117, in pipeline
    .pipe(self.make_monotonic)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 1 chunks of lengths 225 would be NaNed for metric total_vaccinations. That is more than maximum allowed (10) by make_monotonic() - check the data. Check dates ['2023-05-20']. Location(s) affected: {'Hong Kong'}
",1 chunks of lengths 225 would be NaNed for metric total_vaccinations. That is more than maximum allowed (10) by make_monotonic() - check the data. Check dates ['2023-05-20']. Location(s) affected: {'Hong Kong'}
cowidev.vax.batch.indonesia,,,2023-06-11T07:00:06,,
cowidev.vax.batch.ireland,0.4,False,2023-06-11T07:02:02,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/ireland.py"", line 135, in main
    Ireland().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/ireland.py"", line 130, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/ireland.py"", line 27, in read
    return pd.merge(data_primary, data_booster, how=""outer"", on=""date"", validate=""one_to_one"")
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/reshape/merge.py"", line 106, in merge
    op = _MergeOperation(
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/reshape/merge.py"", line 709, in __init__
    self._validate(validate)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/reshape/merge.py"", line 1429, in _validate
    raise MergeError(
pandas.errors.MergeError: Merge keys are not unique in right dataset; not a one-to-one merge
",Merge keys are not unique in right dataset; not a one-to-one merge
cowidev.vax.batch.israel,10.06,True,2023-06-11T07:00:16,,
cowidev.vax.batch.italy,10.37,True,2023-06-11T07:00:16,,
cowidev.vax.batch.japan,9.15,False,2023-06-11T07:02:11,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 375, in main
    Japan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 359, in export
    df = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 71, in read
    df_latest = self.read_latest().pipe(self.pipe_read_latest)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 100, in read_latest
    self._read_latest_booster(),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 211, in _read_latest_booster
    df = self._read_xlsx(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/japan.py"", line 233, in _read_xlsx
    raise ValueError(f""Unknown sheets: {sheets_unknown}"")
ValueError: Unknown sheets: {'５回目_総接種回数', '5.8以降接種_総接種回数', '６回目_総接種回数', '5.8以降接種_一般接種（３回目）', '5.8以降接種_一般接種（４回目）', '５回目_一般接種回数', '5.8以降接種_一般接種（６回目）', '６回目_一般接種回数', '5.8以降接種_一般接種（５回目）'}
","Unknown sheets: {'５回目_総接種回数', '5.8以降接種_総接種回数', '６回目_総接種回数', '5.8以降接種_一般接種（３回目）', '5.8以降接種_一般接種（４回目）', '５回目_一般接種回数', '5.8以降接種_一般接種（６回目）', '６回目_一般接種回数', '5.8以降接種_一般接種（５回目）'}"
cowidev.vax.batch.jersey,5.62,True,2023-06-11T07:00:11,,
cowidev.vax.batch.latvia,51.65,True,2023-06-11T07:00:57,,
cowidev.vax.batch.lithuania,5.47,True,2023-06-11T07:00:11,,
cowidev.vax.batch.luxembourg,3.21,True,2023-06-11T07:00:09,,
cowidev.vax.batch.malaysia,0.58,True,2023-06-11T07:00:06,,
cowidev.vax.batch.malta,0.99,True,2023-06-11T07:00:07,,
cowidev.vax.batch.netherlands,,,2023-06-11T07:00:06,,
cowidev.vax.batch.new_zealand,15.52,True,2023-06-11T07:00:21,,
cowidev.vax.batch.norway,0.35,True,2023-06-11T07:00:06,,
cowidev.vax.batch.peru,5.12,True,2023-06-11T07:00:11,,
cowidev.vax.batch.portugal,,,2023-06-11T07:00:06,,
cowidev.vax.batch.romania,31.58,True,2023-06-11T07:00:37,,
cowidev.vax.batch.saudi_arabia,3.94,True,2023-06-11T07:00:10,,
cowidev.vax.batch.singapore,4.4,True,2023-06-11T07:00:10,,
cowidev.vax.batch.slovakia,45.02,True,2023-06-11T07:00:51,,
cowidev.vax.batch.slovenia,2.25,True,2023-06-11T07:00:08,,
cowidev.vax.batch.south_korea,6.75,False,2023-06-11T07:02:18,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 209, in main
    SouthKorea().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 191, in export
    df_base = self.read().pipe(self.pipeline_base)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 109, in pipeline_base
    .pipe(self.pipe_check_metrics)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/south_korea.py"", line 92, in pipe_check_metrics
    raise ValueError(f""Metric {dose} for 'all' is not equal to the sum over all vaccines (2% tolerance)."")
ValueError: Metric dose_1 for 'all' is not equal to the sum over all vaccines (2% tolerance).
",Metric dose_1 for 'all' is not equal to the sum over all vaccines (2% tolerance).
cowidev.vax.batch.spc,30.56,True,2023-06-11T07:00:36,,
cowidev.vax.batch.sweden,13.47,True,2023-06-11T07:00:20,,
cowidev.vax.batch.switzerland,57.64,True,2023-06-11T07:01:04,,
cowidev.vax.batch.trinidad_and_tobago,5.02,True,2023-06-11T07:00:11,,
cowidev.vax.batch.ukraine,,,2023-06-11T07:00:07,,
cowidev.vax.batch.united_kingdom,4.25,False,2023-06-11T07:02:22,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/united_kingdom.py"", line 156, in main
    UnitedKingdom().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/united_kingdom.py"", line 144, in export
    raise (e)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/batch/united_kingdom.py"", line 141, in export
    df = df.pipe(self.make_monotonic, max_removed_rows=max_removed_rows_dict[location])
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/utils/base.py"", line 120, in make_monotonic
    return mkm(
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 23, in make_monotonic
    return make_monotonic_new(df, column_date, column_metrics, max_removed_rows)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/utils/utils.py"", line 96, in make_monotonic_new
    raise Exception(
Exception: 1 chunks of lengths 261 would be NaNed for metric total_boosters. That is more than maximum allowed (10) by make_monotonic() - check the data. Check dates ['2023-05-31']. Location(s) affected: {'England'}
",1 chunks of lengths 261 would be NaNed for metric total_boosters. That is more than maximum allowed (10) by make_monotonic() - check the data. Check dates ['2023-05-31']. Location(s) affected: {'England'}
cowidev.vax.batch.united_states,23.51,True,2023-06-11T07:00:30,,
cowidev.vax.batch.uruguay,7.94,True,2023-06-11T07:00:16,,
cowidev.vax.batch.zimbabwe,1.93,True,2023-06-11T07:00:11,,
cowidev.vax.incremental.africacdc,,,2023-06-11T07:00:10,,
cowidev.vax.incremental.antigua_barbuda,,,2023-06-11T07:00:10,,
cowidev.vax.incremental.aruba,1.62,True,2023-06-11T07:00:11,,
cowidev.vax.incremental.austria,,,2023-06-11T07:00:10,,
cowidev.vax.incremental.azerbaijan,,,2023-06-11T07:00:10,,
cowidev.vax.incremental.bahrain,,,2023-06-11T07:00:10,,
cowidev.vax.incremental.bangladesh,41.39,True,2023-06-11T07:00:51,,
cowidev.vax.incremental.barbados,,,2023-06-11T07:00:10,,
cowidev.vax.incremental.brazil,1.27,True,2023-06-11T07:00:11,,
cowidev.vax.incremental.bulgaria,7.19,True,2023-06-11T07:00:18,,
cowidev.vax.incremental.china,17.85,True,2023-06-11T07:00:29,,
cowidev.vax.incremental.costa_rica,,,2023-06-11T07:00:11,,
cowidev.vax.incremental.croatia,0.94,True,2023-06-11T07:00:12,,
cowidev.vax.incremental.cuba,1.69,False,2023-06-11T07:02:23,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 54, in _parse_metrics
    data[metric] = clean_count(match.group(1).replace("" "", """"))
AttributeError: 'NoneType' object has no attribute 'group'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 82, in main
    Cuba().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 68, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 29, in read
    return self._parse_data(soup)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 35, in _parse_data
    **self._parse_metrics(soup),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/cuba.py"", line 56, in _parse_metrics
    raise ValueError(
ValueError: Could not parse `people_vaccinated` using regex 'al menos una dosis de una de las vacunas cubanas SOBERANA 02, SOBERANA Plus y ABDALA, ([\d ]+) personas' from website https://salud.msp.gob.cu/actualizacion-de-la-vacunacion-en-el-marco-de-los-estudios-de-los-candidatos-vacunales-cubanos-y-la-intervencion-sanitaria/
","Could not parse `people_vaccinated` using regex 'al menos una dosis de una de las vacunas cubanas SOBERANA 02, SOBERANA Plus y ABDALA, ([\d ]+) personas' from website https://salud.msp.gob.cu/actualizacion-de-la-vacunacion-en-el-marco-de-los-estudios-de-los-candidatos-vacunales-cubanos-y-la-intervencion-sanitaria/"
cowidev.vax.incremental.curacao,,,2023-06-11T07:00:11,,
cowidev.vax.incremental.cyprus,3.94,True,2023-06-11T07:00:15,,
cowidev.vax.incremental.dominican_republic,12.59,False,2023-06-11T07:02:36,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 112, in main
    DominicanRepublic().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 68, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 26, in read
    metrics = self._parse_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 35, in _parse_metrics
    metric_candidates = _find_potential_metrics(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 91, in _find_potential_metrics
    h3 = _find_h3(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/dominican_republic.py"", line 81, in _find_h3
    raise ValueError(""More than one element found!"")
ValueError: More than one element found!
",More than one element found!
cowidev.vax.incremental.el_salvador,8.13,True,2023-06-11T07:00:19,,
cowidev.vax.incremental.faeroe_islands,2.39,True,2023-06-11T07:00:14,,
cowidev.vax.incremental.fiji,0.0,True,2023-06-11T07:00:12,,
cowidev.vax.incremental.finland,1.72,True,2023-06-11T07:00:13,,
cowidev.vax.incremental.gabon,,,2023-06-11T07:00:13,,
cowidev.vax.incremental.georgia,4.47,True,2023-06-11T07:00:18,,
cowidev.vax.incremental.greenland,,,2023-06-11T07:00:14,,
cowidev.vax.incremental.guatemala,,,2023-06-11T07:00:14,,
cowidev.vax.incremental.guernsey,,,2023-06-11T07:00:14,,
cowidev.vax.incremental.hungary,78.0,True,2023-06-11T07:01:32,,
cowidev.vax.incremental.iceland,1.29,True,2023-06-11T07:00:16,,
cowidev.vax.incremental.india,1.59,True,2023-06-11T07:00:17,,
cowidev.vax.incremental.iran,,,2023-06-11T07:00:16,,
cowidev.vax.incremental.isle_of_man,2.92,True,2023-06-11T07:00:19,,
cowidev.vax.incremental.jamaica,8.01,True,2023-06-11T07:00:24,,
cowidev.vax.incremental.kazakhstan,8.3,True,2023-06-11T07:00:24,,
cowidev.vax.incremental.kosovo,,,2023-06-11T07:00:16,,
cowidev.vax.incremental.kyrgyzstan,4.33,True,2023-06-11T07:00:20,,
cowidev.vax.incremental.laos,1.64,True,2023-06-11T07:02:38,,
cowidev.vax.incremental.macao,6.59,False,2023-06-11T07:02:44,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 90, in main
    Macao().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 75, in export
    data = self.read()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 23, in read
    iframe_url = self._get_iframe_url(driver)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/macao.py"", line 31, in _get_iframe_url
    elem = driver.find_element_by_id(""ICovid19Monitor"")
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 360, in find_element_by_id
    return self.find_element(by=By.ID, value=id_)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 976, in find_element
    return self.execute(Command.FIND_ELEMENT, {
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/webdriver.py"", line 321, in execute
    self.error_handler.check_response(response)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/selenium/webdriver/remote/errorhandler.py"", line 242, in check_response
    raise exception_class(message, screen, stacktrace)
selenium.common.exceptions.NoSuchElementException: Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=114.0.5735.106)

","Message: no such element: Unable to locate element: {""method"":""css selector"",""selector"":""[id=""ICovid19Monitor""]""}
  (Session info: headless chrome=114.0.5735.106)
"
cowidev.vax.incremental.moldova,,,2023-06-11T07:00:17,,
cowidev.vax.incremental.monaco,80.17,True,2023-06-11T07:01:38,,
cowidev.vax.incremental.mongolia,1.93,True,2023-06-11T07:00:20,,
cowidev.vax.incremental.montenegro,0.95,True,2023-06-11T07:00:19,,
cowidev.vax.incremental.morocco,,,2023-06-11T07:00:18,,
cowidev.vax.incremental.myanmar,,,2023-06-11T07:00:18,,
cowidev.vax.incremental.nepal,,,2023-06-11T07:00:18,,
cowidev.vax.incremental.north_macedonia,,,2023-06-11T07:00:18,,
cowidev.vax.incremental.northern_cyprus,4.69,True,2023-06-11T07:00:23,,
cowidev.vax.incremental.paho,82.6,False,2023-06-11T07:04:07,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 195, in main
    PAHO().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 190, in export
    df = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 167, in pipeline
    df.pipe(self.pipe_check_columns)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/generic.py"", line 5430, in pipe
    return com.pipe(self, func, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/venv/lib/python3.10/site-packages/pandas/core/common.py"", line 471, in pipe
    return func(obj, *args, **kwargs)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/paho.py"", line 122, in pipe_check_columns
    raise ValueError(f""Missing column fields: {columns_missing}. Present columns are: {df.columns}"")
ValueError: Missing column fields: {'2nd additional dose', 'First dose', 'Total doses', 'Single dose', 'Country/ Territory', 'Country code', 'Second dose', '1st additional dose'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')
","Missing column fields: {'2nd additional dose', 'First dose', 'Total doses', 'Single dose', 'Country/ Territory', 'Country code', 'Second dose', '1st additional dose'}. Present columns are: Index(['date,location,total_vaccinations,people_vaccinated,people_fully_vaccinated,total_boosters,new_vaccinations,new_vaccinations_interpolated,new_people_vaccinated_interpolated,new_vaccinations_smoothed,new_people_vaccinated_smoothed', 'date'], dtype='object')"
cowidev.vax.incremental.pakistan,,,2023-06-11T07:00:19,,
cowidev.vax.incremental.philippines,,,2023-06-11T07:00:19,,
cowidev.vax.incremental.poland,1.73,True,2023-06-11T07:00:21,,
cowidev.vax.incremental.qatar,,,2023-06-11T07:00:19,,
cowidev.vax.incremental.russia,19.76,True,2023-06-11T07:00:39,,
cowidev.vax.incremental.saint_lucia,,,2023-06-11T07:00:20,,
cowidev.vax.incremental.serbia,9.98,True,2023-06-11T07:00:30,,
cowidev.vax.incremental.south_africa,,,2023-06-11T07:00:20,,
cowidev.vax.incremental.spain,11.91,True,2023-06-11T07:00:32,,
cowidev.vax.incremental.sri_lanka,,,2023-06-11T07:00:20,,
cowidev.vax.incremental.suriname,14.6,True,2023-06-11T07:00:35,,
cowidev.vax.incremental.taiwan,5.97,False,2023-06-11T07:04:13,"Traceback (most recent call last):
  File ""/home/owid/covid-19-data/scripts/src/cowidev/cmd/commons/get.py"", line 47, in run
    module.main()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 184, in main
    Taiwan().export()
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 170, in export
    data = self.read().pipe(self.pipeline)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 38, in read
    data = self.parse_data(df, soup)
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 122, in parse_data
    ""date"": self._parse_date(soup),
  File ""/home/owid/covid-19-data/scripts/src/cowidev/vax/incremental/taiwan.py"", line 156, in _parse_date
    date_str = re.search(regex, date_raw).group(1)
AttributeError: 'NoneType' object has no attribute 'group'
",'NoneType' object has no attribute 'group'
cowidev.vax.incremental.thailand,5.8,True,2023-06-11T07:00:26,,
cowidev.vax.incremental.turkey,,,2023-06-11T07:00:21,,
cowidev.vax.incremental.united_arab_emirates,,,2023-06-11T07:00:21,,
cowidev.vax.incremental.vietnam,,,2023-06-11T07:00:21,,
cowidev.vax.incremental.who,24.44,True,2023-06-11T07:00:46,,
cowidev.vax.incremental.zambia,1.64,True,2023-06-11T07:00:24,,
